abnf/odata-aggregation-abnf.txt

```diff
@@ -137,16 +137,16 @@ customAggregate = odataIdentifier
 ; - one for use in aggregate, whose segments can be single- or collection-valued (rules with prefix aggr)
 ; - one for use in groupby, whose segments must be single-valued (rules with prefix sngl)
 ; - one for use in transformnested and addnested, without entity-valued segments (rule with prefix nest)
-; A type cast (optionallyQualifiedComplexTypeName or optionallyQualifiedEntityTypeName) cannot be the final segment in them.
 ; Term casts are not allowed in data aggregation paths.
-aggrPropPath = ( complexProperty          / complexColProperty          ) [ [ "/" optionallyQualifiedComplexTypeName ] "/" aggrPropPath ]
-             / ( entityNavigationProperty / entityColNavigationProperty ) [ [ "/" optionallyQualifiedEntityTypeName  ] "/" aggrPropPath ]
-aggrPrimPath = ( complexProperty          / complexColProperty          )   [ "/" optionallyQualifiedComplexTypeName ] "/" aggrPrimPath
-             / ( entityNavigationProperty / entityColNavigationProperty )   [ "/" optionallyQualifiedEntityTypeName  ] "/" aggrPrimPath
-             /   primitiveProperty        / primitiveColProperty
-             /   streamProperty
-             
-nestPropPath = ( complexProperty          / complexColProperty          ) [ [ "/" optionallyQualifiedComplexTypeName ] "/" nestPropPath ]
+aggrPropStep = ( complexProperty          / complexColProperty          ) [ "/" optionallyQualifiedComplexTypeName ]
+             / ( entityNavigationProperty / entityColNavigationProperty ) [ "/" optionallyQualifiedEntityTypeName  ]
+aggrPropPath = aggrPropStep [ "/" aggrPropPath ]
+aggrPrimPath = aggrPropStep "/" aggrPrimPath
+             / primitiveProperty / primitiveColProperty / streamProperty
+aggrCastPath = optionallyQualifiedComplexTypeName / optionallyQualifiedEntityTypeName
+optionalLeadingTypeCast = [ aggrCastPath "/" ]

```

Comment:
This re-introduces issue #28, better inline this expression (which is actually shorter than its "abbreviation").